frontend/src/pages/blog/Article.js

```diff
@@ -4,9 +4,28 @@ import { useArticlesContext } from "../../hooks/useArticlesContext";
 
 const Article = () => {
   let { id } = useParams();
-  const { articles } = useArticlesContext();
-  const article = articles.filter((a) => a._id === id)[0];
-  return <ArticleView article={article} />;
+  const { articles, dispatch } = useArticlesContext();
+
+  const fetchArticle = async () => {
+      const response = await fetch(`/api/articles/${id}`);  // fetch the article (only this one) from the database
+      if (response.ok) {
+        const article = await response.json();
+        dispatch({ type: "ADD", payload: article });       // if it exists, add it to the ArticlesContext
+        return article;
+      }
+      else {
+        return null;
+      } 
+  };
+  
+  try {
```

Comment:
Just some minor comments here:
- I would use `let` instead of `const`
- You should move this return outside of the try/catch
- I found a cleaner way to do `filter(...)[0]`

```js
let article;
try {
    article = articles.find((a) => a._id === id);
} catch (error) {  //if articles are null, articles.filter produces an error
    article = fetchArticle();
}

return <ArticleView article={article} />;
```

Reply:
Let's say that my "limited `JS` knowledge created this infamy 😅 fixed in 0d870937c5d299647f9a40d996205c7e7020bdf0

Reply:
Perfect!

Reply:
Actually I'm wondering: why don't we need to pass `id` as parameter of `fetchArticle`?

Reply:
`id`is defined within the scop of the `Article` class (see line 1) so it already sees it

Reply:
I get it but it feels weird I guess!